service/policy/db/resource_mapping.go

```diff
@@ -199,7 +319,73 @@ func (c PolicyDBClient) ListResourceMappings(ctx context.Context) ([]*policy.Res
 	return list, nil
 }
 
-func updateResourceMappingSQL(id string, attributeValueID string, metadata []byte, terms []string) (string, []interface{}, error) {
+// NOTE: uses sqlc instead of squirrel
+func (c PolicyDBClient) ListResourceMappingsByGroupFqns(ctx context.Context, fqns []string) (map[string]*resourcemapping.ResourceMappingsByGroup, error) {
+	resp := make(map[string]*resourcemapping.ResourceMappingsByGroup)
+	resultCount := 0
+
+	for _, fqn := range fqns {
+		fullyQualifiedGroup, err := util.ParseResourceMappingGroupFqn(fqn)
+		if err != nil {
+			// invalid FQNs not included in the response - ignore and continue, but log for investigation
+			slog.DebugContext(ctx, "error parsing Resource Mapping Group FQN", slog.String("rmg_fqn", fqn))
+			continue
+		}
+
+		rows, err := c.Queries.ListResourceMappingsByFullyQualifiedGroup(ctx, ListResourceMappingsByFullyQualifiedGroupParams{
+			NamespaceName: fullyQualifiedGroup.Namespace,
+			GroupName:     fullyQualifiedGroup.GroupName,
+		})
+		if err != nil {
+			return nil, db.WrapIfKnownInvalidQueryErr(err)
+		}
+
+		if len(rows) == 0 {
+			// no rows found for this FQN - ignore and continue
+			continue
+		}
+
+		resultCount++
+
+		mappings := make([]*policy.ResourceMapping, len(rows))
+		for i, row := range rows {
+			metadata := new(common.Metadata)
+			if row.Metadata != nil {
+				if err := protojson.Unmarshal(row.Metadata, metadata); err != nil {
+					return nil, err
+				}
+			}
+
+			mappings[i] = &policy.ResourceMapping{
+				Id:             row.ID,
+				AttributeValue: &policy.Value{Id: row.AttributeValueID},
```

Comment:
I think we should handle this as a separate issue and get all your work merged, but I'm wondering if the value's ID alone is sufficient for the response in this flow.